rascal2/widgets/inputs.py

```diff
@@ -8,76 +8,154 @@
 from PyQt6 import QtCore, QtGui, QtWidgets
 
 
-class ValidatedInputWidget(QtWidgets.QWidget):
-    """Value input generated from Pydantic field info.
+def get_validated_input(field_info: FieldInfo) -> QtWidgets.QWidget:
+    """Get a validated input widget from Pydantic field info.
 
     Parameters
     ----------
-    field_info: FieldInfo
+    field_info : FieldInfo
+        The Pydantic field info for the field.
+
+    Returns
+    -------
+    QtWidgets.QWidget
+        The validated input widget for the field.
+
+    """
+    class_widgets = {
+        bool: BoolInputWidget,
+        int: IntInputWidget,
+        float: FloatInputWidget,
+        Enum: EnumInputWidget,
+    }
+
+    for type, widget in class_widgets.items():
```

Comment:
The names `type` and `input` shadow built-in globals while I don't think anyone will every use `input` in this project, I advise changing these names to avoid conflict or bugs in future. Might be good to add this to our linter if its easy to do

Reply:
good spot - i've added the `flake8-naming` ruleset to the linting and fixed a couple other places where it was broken.